src/link.js

```diff
@@ -88,17 +100,23 @@ module.exports = function link(config, args) {
     packageName ? [packageName] : getProjectDependencies()
   );
 
-  const tasks = Promise.all(dependencies.map(dependency => {
-    const pre = promisify(dependency.config.commands.prelink || commandStub);
-    const post = promisify(dependency.config.commands.postlink || commandStub);
-
-    return pre().then(() => linkDependency(project, dependency)).then(post);
-  }));
-
-  const assets = dedupeAssets(dependencies.reduce(
-    (assets, dependency) => assets.concat(dependency.config.assets),
-    project.assets
-  ));
-
-  return tasks.then(() => linkAssets(project, assets));
+  return Promise.all(
+    dependencies.map(dependency => promiseWaterfall([
+      () => pollParams(dependency.config.params),
+      (params) => new Promise((res, rej) => {
+        dependency.config.params = params;
+        res();
+      }),
+      () => promisify(dependency.config.commands.postlink || commandStub),
+      () => linkDependency(project, dependency),
+      () => promisify(dependency.config.commands.prelink || commandStub),
+    ]))
+  ).then(() => {
+    const assets = dedupeAssets(dependencies.reduce(
```

Comment:
Can we maybe make assets above line `103` and make this just `then(() => linkAssets(...))` ?